pandas/core/frame.py

```diff
@@ -4129,14 +4114,16 @@ def set_index(
                     found = col in self.columns
                 except TypeError:
                     raise TypeError(
-                        err_msg + " Received column of type {}".format(type(col))
+                        f"{err_msg} Received column of type {type(col)}"
+                        "array, or a list containing only valid column keys and "
+                        f"one-dimensional arrays. Received column of type {type(col)}"
```

Comment:
why's this changed?

Reply:
My mistake, I misinterpreted a previous comment